web/modules/custom/chroma/src/Plugin/Field/FieldFormatter/ChromaBackgroundFormatter.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\chroma\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Displays the code along with the colour in the background.
+ *
+ * @FieldFormatter(
+ *   id = "chroma_background",
+ *   label = @Translation("View Colour"),
+ *   field_types = {
+ *     "colour"
+ *   },
+ * )
+ */
+class ChromaBackgroundFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'rgb_value' => '',
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $elements['rgb_value'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Show RGB value of colour'),
+      '#default_value' => $this->getSetting('rgb_value'),
+    ];
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = [];
+    $settings = $this->getSettings();
+
+    if ($settings['rgb_value']) {
+      $summary[] = $this->t('Displays the RGB value of colour.');
+    }
+    else {
+      $summary[] = $this->t('Displays the hexcode of colour.');
+    }
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $element = [];
+    $settings = $this->getSettings();
+
+    foreach ($items as $delta => $item) {
+      if ($settings['rgb_value']) {
+        $element[$delta] = [
+          '#type' => 'html_tag',
+          '#tag' => 'div',
+          '#value' => $this->colourFormat($item->value),
+          '#attributes' => [
+            'style' => 'background-color: ' . $item->value . ';',
+          ],
+        ];
+      }
+      else {
+        $element[$delta] = [
+          '#type' => 'html_tag',
+          '#tag' => 'div',
+          '#value' => $item->value,
+          '#attributes' => [
+            'style' => 'background-color: ' . $item->value . ';',
+          ],
+        ];
+      }
```

Comment:
Try to identify repititions and try to refactor it.
Here this can be done like this:
```
        $element[$delta] = [
          '#type' => 'html_tag',
          '#tag' => 'div',
          '#attributes' => [
            'style' => 'background-color: ' . $item->value . ';',
          ],
        ];
   
        $element[$delta]['#value'] = $settings['rgb_value'] 
           ? $this->colourFormat($item->value)
           :  $item->value;
```